code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -1103,6 +1103,21 @@ func (sc *StorageSmartContract) populateGenerateChallenge(
 				return nil, common.NewError("add_challenge",
 					err.Error())
 			}
+
+			actErr = cstate.WithActivation(balances, "demeter", func() error {
+				return nil
+			}, func() error {
+				err = validators.Remove(balances, validator.Id())
```

Comment:
not sure we need to delete here

Reply:
If validator is not in MPT (which can be the casse when validator is killed), we should remove it from the partition as well right? 

I added this because in kill validator we were not removing the validator from partition. I have fixed it now.

Reply:
Fixed here : https://github.com/0chain/0chain/pull/3308/commits/fba7b9d8e9cc3042201a239fa642f9b22399b5b8.